homeassistant/components/universal/media_player.py

```diff
@@ -625,9 +634,15 @@ async def async_toggle(self):
 
     async def async_update(self):
         """Update state in HA."""
-        for child_name in self._children:
-            child_state = self.hass.states.get(child_name)
-            if child_state and child_state.state not in OFF_STATES:
-                self._child_state = child_state
-                return
         self._child_state = None
+        for child_name in self._children:
+            if (child_state := self.hass.states.get(child_name)) and STATES_ORDER.index(
+                child_state.state
+            ) >= STATES_ORDER.index(STATE_ON):
+                if self._child_state:
+                    if STATES_ORDER.index(child_state.state) > STATES_ORDER.index(
+                        self._child_state.state
+                    ):
+                        self._child_state = child_state
+                else:
+                    self._child_state = child_state
```

Comment:
You should be able to use `max` for this. Untested:
```suggestion
        self._child_state = max(
            self._children,
            key=lambda: entity_id: STATES_ORDER.index(state.state) if (state := self.hass.states.get(entity_id) else -1
        )
```

Reply:
Ah that works but it doesn't filter the states.

Reply:
Would `max` be better? Not obvious to me how it would, tbh.

Reply:
Any prospect of getting this moved forward? Does it work?